tests/step_implementers/unit_test/test_junit.py

```diff
@@ -1,24 +1,472 @@
+import unittest
+from unittest.mock import patch
 import yaml
 
 from tssc import TSSCFactory
 from tssc.step_implementers.unit_test import JUnit
 
 from test_utils import *
 
-def test_tag_source_specify_junit_implementer():
-    with TempDirectory() as temp_dir:
+def create_mvn_side_effect(pom_file, artifact_parent_dir, artifact_names, throw_mvn_exception=False):
+    """simulates what mvn does by touching files.
+    
+    Notes
+    -----
+    
+    Supports
+    
+    - mvn clean
+    - mvn install
+    - mvn test
+    
+    """
+    target_dir_path = os.path.join(
+        os.path.dirname(os.path.abspath(pom_file)),
+        artifact_parent_dir)
+
+    
+    def mvn_side_effect(*args, **kwargs):
+        if 'clean' in args:
+            if os.path.exists(target_dir_path):
+                os.rmdir(target_dir_path)
+        
+        if 'install' in args:
+            os.mkdir(target_dir_path)
+            
+            for artifact_name in artifact_names:
+                artifact_path = os.path.join(
+                    target_dir_path,
+                    artifact_name
+                )
+                Path(artifact_path).touch()
+        
+        if 'test' in args:
+            if throw_mvn_exception is True:
+                raise RuntimeError
+
+            os.makedirs(target_dir_path, exist_ok=True)
+            
+            for artifact_name in artifact_names:
+                artifact_path = os.path.join(
+                    target_dir_path,
+                    artifact_name
+                )
+                Path(artifact_path).touch()
+
+    return mvn_side_effect
+
+class TestStepImplementerUnitTest(unittest.TestCase):
+
+    @patch('sh.mvn', create=True)
+    def test_unit_test_default_pom_file_missing(self, mvn_mock):
+        config = {
+            'tssc-config': {
+                'unit-test': {
+                    'implementer': 'JUnit'
+                }
+            }
+        }
+        factory = TSSCFactory(config)
+        with self.assertRaisesRegex(
+                ValueError, 
+                "Given pom file does not exist: pom.xml"):
+            factory.run_step('unit-test')
+
+    @patch('sh.mvn', create=True)
+    def test_unit_test_runtime_pom_file_missing(self, mvn_mock):
+        config = {
+            'tssc-config': {
+                'unit-test': {
+                    'implementer': 'JUnit'
+                }
+            }
+        }
+        factory = TSSCFactory(config)
+        with self.assertRaisesRegex(
+                ValueError, 
+                "Given pom file does not exist: does-not-exist-pom.xml"):
+            factory.run_step('unit-test', {'pom-file': 'does-not-exist-pom.xml'})
+    
+    @patch('sh.mvn', create=True)
+    def test_unit_test_config_file_pom_file_missing(self, mvn_mock):
         config = {
-            'tssc-config': {    
+            'tssc-config': {
                 'unit-test': {
                     'implementer': 'JUnit',
```

Comment:
I think we need to decide on a standard capitalization scheme for the implementer names

Reply:
I agree. I noticed that others were using capitalization, so I followed suit with how JUnit is normally capitalized. My preference is none. I've created the following task to clean everything up and added it to the backlog: https://projects.engineering.redhat.com/browse/NAPSSPO-786